### PR TITLE
Make Future generic

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement
Also add `Go` method to use future without `Pool`

### CODE REVIEW

File: internal/util/concurrency/pool.go [hunk 1]
```diff
@@ -39,8 +39,9 @@ func NewPool(cap int, opts ...ants.Option) (*Pool, error) {
 // Submit a task into the pool,
 // executes it asynchronously.
 // This will block if the pool has finite workers and no idle worker.
-func (pool *Pool) Submit(method func() (interface{}, error)) *Future {
-	future := newFuture()
+// NOTE: As now golang doesn't support the member method being generic, we use Future[any]
+func (pool *Pool) Submit(method func() (any, error)) *Future[any] {
```
Comment (congqixia) [inline internal/util/concurrency/pool.go:43]: Why not make `Pool` generic? If the return type is different for a pool, user could choose to use `any` for generic type.

Comment (yah01) [reply internal/util/concurrency/pool.go:43]: make sense

Comment (congqixia) [reply internal/util/concurrency/pool.go:43]: This could be done by manually specify by using `Pool[any]`

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**